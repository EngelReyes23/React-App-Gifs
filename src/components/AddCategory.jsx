import React, { useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({ setCategories }) => {
	// Estado del input
	const [inputValue, setInputValue] = useState("");

	//
	const handleInputChange = (e) => {
		setInputValue(e.target.value);
	};

	//
	const handleSubmit = (e) => {
		e.preventDefault();

		// Verifica que inputValue no esté vacío
		if (inputValue !== "") {
			setCategories((categories) => [inputValue, ...categories]);
			setInputValue("");
		}
	};

	return (
		<form onSubmit={handleSubmit}>
			<input
				type="text"
				onChange={handleInputChange}
				placeholder="Busca Algo"
				value={inputValue}
			/>
		</form>
	);
};

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
