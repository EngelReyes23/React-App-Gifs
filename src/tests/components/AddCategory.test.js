import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe("Pruebas al component <AddCategory />", () => {
	// simula el argumento de funcion que requiere el componente
	const setCategories = jest.fn();
	// simula el argumento de funcion que requiere el componente
	let wrapper;

	// beforeEach limpia todo el rastro de eventos
	beforeEach(() => {
		jest.clearAllMocks();
		wrapper = shallow(<AddCategory setCategories={setCategories} />);
	});

	// 1ra prueba
	test("debe mostrar correctamente el componente", () => {
		expect(wrapper).toMatchSnapshot();
	});

	//
	// test("debe cambiar la caja de texto", () => {
	// 	const input = wrapper.find("input");
	// 	input.simulate("change", { target: { value: "Hola" } });
	// 	expect(input.text().trim()).toBe("Hola");
	// });

	// 2da prueba
	test("no debe postear la informacion on Submit", () => {
		wrapper.find("form").simulate("submit", { preventDefault() {} });
		expect(setCategories).not.toHaveBeenCalled();
	});

	// 3ra prueba
	test("debe de llamar al setCategories y limpiar la caja de texto", () => {
		/**
		 * 1- Simula el inputChange
		 * 2- Simula el submit
		 * 3- setCategories se debe de haber llamado
		 * 4. el valor del input debe de estar vacio
		 */

		// Simula el input change
		wrapper.find("input").simulate("change", { target: { value: "Hola" } });

		// Simula el submit
		wrapper.find("form").simulate("submit", { preventDefault() {} });

		// verifica si setCategories fue llamado
		expect(setCategories).toHaveBeenCalled();

		// verifica si el valor del input esta vacio
		expect(wrapper.find("input").prop("value")).toBe("");
	});
});
