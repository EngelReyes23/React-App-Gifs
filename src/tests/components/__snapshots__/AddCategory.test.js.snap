// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al component <AddCategory /> debe mostrar correctamente el componente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Busca Algo..."
    type="text"
    value=""
  />
</form>
`;
